[Merged by Bors] - chore: auto `var decl var` replacements

adomani opened the pull request:
Found by #22389.

The motivation comes from [this Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/style.20proposal.3A.20avoid.20.60variable.20.7BX.7D.20.2E.2E.2E.20variable.20.28X.29.60).

Summarising, updating a variable binder for a single declaration can be confusing.

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

adomani commented on Mathlib/Algebra/Star/Basic.lean:
```diff
@@ -299,7 +293,8 @@ theorem star_intCast [Ring R] [StarRing R] (z : ℤ) : star (z : R) = z :=
 section CommSemiring
-variable [CommSemiring R] [StarRing R]
+
+variable [CommSemiring R] [StarRing R] in
```
Don't know why this happened...

grunweg reviewed (approved):
Thanks; all changes look good to me. Can you add a brief word on the motivation to the introduction?

I have not verified that all modified variables were really implicit (as opposed to instance-implicit); I trust your linter in this regard.
maintainer delegate

adomani commented:
I updated the PR description.  The linter only flags switches between implicit/explicit, in either direction, and ignores all other binder types.

I also updated the PR description.

j-loreaux commented on Mathlib/Algebra/MvPolynomial/Supported.lean:
```diff
@@ -30,14 +30,13 @@ variable {σ : Type*} {R : Type u}
-variable (R)
+
+variable (R) in
 
```
so that it abuts the declaration.
```suggestion
```

j-loreaux commented on Mathlib/GroupTheory/GroupAction/Defs.lean:
```diff
@@ -528,7 +525,8 @@ lemma univ_eq_iUnion_orbit :
-variable (G)
+
+variable (G) in
 
```
```suggestion
```

j-loreaux commented on Mathlib/RingTheory/TensorProduct/Basic.lean:
```diff
@@ -59,16 +59,15 @@ variable [Semiring A] [Algebra R A] [Semiring B] [Algebra R B]
-variable (A)
+
+variable (A) in
 
```
```suggestion
```

j-loreaux reviewed (commented):
Thanks!

bors d+

adomani commented:
bors merge
